azext_iot/deviceupdate/providers/base.py

```diff
@@ -0,0 +1,190 @@
+
+        # Adding IoT Ext User-Agent is done with best attempt.
+        try:
+            client._config.user_agent_policy.add_user_agent(USER_AGENT)
+        except Exception:
+            pass
+
+        return client
+
+    def get_data_client(self):
+        pass
+
+
+class DeviceUpdateAccountManager(DeviceUpdateClientHandler):
+    def __init__(self, cmd):
+        super().__init__(cmd=cmd)
+        self.mgmt_client = self.get_mgmt_client()
+        self.cli = EmbeddedCLI()
+
+    def find_account(self, target_name: str, target_rg: str = None) -> AccountContainer:
+        def find_account_rg(id: str):
+            return id.split("/")[4]
+
+        if target_rg:
+            account = self.mgmt_client.accounts.get(
+                resource_group_name=target_rg, account_name=target_name
+            )
+            return AccountContainer(account, find_account_rg(account.id))
+
+        for account in self.mgmt_client.accounts.list_by_subscription():
+            if account.name == target_name:
+                return AccountContainer(account, find_account_rg(account.id))
+
+        raise ResourceNotFoundError(
+            f"DeviceUpdate account: '{target_name}' not found by auto-discovery. "
+            "Provide resource group via -g for direct lookup."
+        )
+
+    @classmethod
+    def assemble_account_auth(
+        cls,
+        assign_identity: list,
+    ) -> Union[None, DeviceUpdateMgmtModels.ManagedServiceIdentity]:
+        if not assign_identity:
+            return None
+
+        if len(assign_identity) == 1:
+            if SYSTEM_IDENTITY_ARG in assign_identity:
+                return DeviceUpdateMgmtModels.ManagedServiceIdentity(
+                    type=DeviceUpdateMgmtModels.ManagedServiceIdentityType.SYSTEM_ASSIGNED
+                )
+            else:
+                return DeviceUpdateMgmtModels.ManagedServiceIdentity(
+                    type=DeviceUpdateMgmtModels.ManagedServiceIdentityType.USER_ASSIGNED,
+                    user_assigned_identities={assign_identity[0], {}},
+                )
+        else:
+            target_identity_type = (
+                DeviceUpdateMgmtModels.ManagedServiceIdentityType.USER_ASSIGNED
+            )
+            user_assigned_identities = {}
+            has_system = False
+            for identity in assign_identity:
+                if identity == SYSTEM_IDENTITY_ARG and not has_system:
+                    target_identity_type = (
+                        DeviceUpdateMgmtModels.ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED
+                    )
+                    has_system = True
+                else:
+                    user_assigned_identities[identity] = {}
+
+            return DeviceUpdateMgmtModels.ManagedServiceIdentity(
+                type=target_identity_type,
+                user_assigned_identities=user_assigned_identities,
+            )
+
+    def assign_msi_scope(
```

Comment:
this and the group location can be generalized since they are similarly used in other commands

Reply:
Sure. Made a task to rack, but not doing in this PR.